sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionInitializingEventArgs.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Contains information about a partition that an <see cref="EventProcessorClient" /> will be
+    ///   processing events from.  It can also be used to specify the position within a partition
+    ///   where the associated <see cref="EventProcessorClient" /> should begin reading events in case
+    ///   it cannot find a checkpoint.
+    /// </summary>
+    ///
+    public class PartitionInitializingEventArgs
+    {
+        /// <summary>
+        ///   The context of the Event Hub partition this instance is associated with.
+        /// </summary>
+        ///
+        public PartitionContext Partition { get; }
```

Comment:
In light of Srikanta's point during discussion yesterday, should we consider making this just the Id, since the partition is not initialized and doesn't really have an active context?

Reply:
I think it makes sense. The partition id is the only thing we're passing when starting because we don't have a consumer to do so.

Reply:
Not 100% related, but I just realized something:
- A `PartitionContext` is created (without a consumer) before initializing. This context is stored.
- When processing events and stopping, we pass them the stored context (without the inner consumer). We shouldn't be doing that.

Like I said, it's not exactly related. We have this problem now and we'll still have it if we don't include a `PartitionContext` in `PartitionInitializingEventArgs`.

https://github.com/Azure/azure-sdk-for-net/blob/68b1a07746e77556f7c5bf64f5f387acce37b747/sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs#L1207-L1208

That's easy to solve for the `OnProcessEventAsync` call. We just need to pass the context in the `partitionEvent` instead.

https://github.com/Azure/azure-sdk-for-net/blob/68b1a07746e77556f7c5bf64f5f387acce37b747/sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs#L1073-L1078

That's not that trivial for `OnPartitionClosing`. I believe the context should contain the inner consumer so users can retrieve the last received event properties. We could store it in a `ConcurrentDictionary` every time we receive an event so we can pass it to the handler when closing, but I think that's going to result in a performance drop.

Reply:
Also, just realized we can't reuse the context passed to `OnPartitionInitializingAsync` exactly because it lacks the inner consumer. We would need to use other instances for processing and stopping.

Reply:
Not sure that I'm following completely.   I think we agree that this should change to `PartitionId` and not pass the full context.     Processing events would just use the context from the `PartitionEvent` received by the iterator; that manages sharing already.    The close args would probably need to do something like keep track of the last event for the partition...   that we should probably talk over and see if it's worth the effort versus just returning the partition id there.

Reply:
Just updated `PartitionInitializingEventArgs` to take a `partitionId` instead of a context.

It seems the `partitionEvent.Partition` received from `ReadEventsFromPartitionAsync` is `null` sometimes, which was causing some issues. I added a small `if` to handle that for the time being.

Regarding the context we pass to `PartitionClosingEventArgs`, I'm currently storing the last received context everytime we receive something, which is very bad (specially because it stores it in a `ConcurrentDictionary`). We may need to come up with a workaround for that.